core-tests/shared/src/test/scala/zio/ZIOSpec.scala

```diff
@@ -3121,6 +3123,16 @@ object ZIOSpec extends ZIOBaseSpec {
           b      <- effect.await
         } yield assert(b)(equalTo(42))
       } @@ zioTag(interruption),
+      test("raceFirst always calls finalizers even with blocking operation") {
+        for {
+          isRunning              <- ZIO.succeed(new AtomicBoolean(true))
+          backgroundBlockingStuff = ZIO.attemptBlocking(while (isRunning.get()) {})
+          acquire                 = backgroundBlockingStuff.fork
+          release                 = ZIO.succeed(isRunning.set(false))
+          eff                     = ZIO.acquireRelease(acquire)(_ => release)
+          _                      <- ZIO.scoped(ZIO.succeed(ZIO.raceFirst(eff, Nil)))
```

Comment:
This test if wrong because `raceFirst` is not evaluated. You need to remove `ZIO.succeed`